File: frontend/src/types/telegram.ts
/**
 * Telegram bot integration types.
 */

/** Telegram link status. */
export interface TelegramStatus {
  is_linked: boolean
  telegram_username: string | null
  notify_deadlines: boolean
  morning_summary: boolean
  link_code: string | null
  link_code_expires_at: string | null
}

/** Response from link code generation. */
export interface TelegramLinkCode {
  link_code: string
  expires_at: string
  bot_username: string
}

/** Notification preferences update. */
export interface TelegramNotificationsUpdate {
  notify_deadlines?: boolean
  morning_summary?: boolean
}

File: backend/src/schemas/telegram.py
"""Telegram bot schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class TelegramStatusResponse(BaseModel):
    """Telegram link status for the current user."""

    is_linked: bool
    telegram_username: str | None = None
    notify_deadlines: bool = True
    morning_summary: bool = True
    link_code: str | None = None
    link_code_expires_at: datetime | None = None


class TelegramLinkCodeResponse(BaseModel):
    """Response with a generated link code."""

    link_code: str
    expires_at: datetime
    bot_username: str


class TelegramNotificationsUpdate(BaseModel):
    """Update notification preferences."""

    notify_deadlines: bool | None = Field(None, description="Toggle deadline alerts")
    morning_summary: bool | None = Field(None, description="Toggle morning summary")


class TelegramLinkResponse(BaseModel):
    """Full telegram link response (admin/internal)."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    telegram_id: int | None
    telegram_username: str | None
    notify_deadlines: bool
    morning_summary: bool
    linked_at: datetime | None
    created_at: datetime
    updated_at: datetime
